framework/src/utils/MooseMeshUtils.C

```diff
@@ -61,6 +61,9 @@ changeBoundaryId(MeshBase & mesh,
   // from showing up when printing information, etc.
   if (delete_prev)
     boundary_info.remove_id(old_id);
+
+  // global information may now be out of sync
+  mesh.set_isnt_prepared();
```

Comment:
depends if we re calling this in parallel or not

do we have such a boolean to check?

changeSubdomainId does not do this

Reply:
`changeSubdomainId` should also `set_isnt_prepared` since you may also have possibly invalidated the global subdomain ID cache

We do not have a boolean to know whether everyone is calling this method. Do we want to make that a requirement?

Reply:
want to fix changeSubdomainID then? same file

Reply:
> no I would only want this to be a requirement if we observe that this "set_isnt_prepared" is making us slower again

Yea it's a good comment. I've opened https://github.com/libMesh/libmesh/pull/3636 in that vein.

> want to fix changeSubdomainID then? same file

Yes I will do that